exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java

```diff
@@ -190,22 +136,40 @@ private static String commonPath(FileStatus... paths) {
         break;
       }
     }
-    URI oneURI = paths[0].getPath().toUri();
+    URI oneURI = statuses.get(0).getPath().toUri();
     return new Path(oneURI.getScheme(), oneURI.getAuthority(), commonPath).toString();
   }
 
-  public static FileSelection create(DrillFileSystem fs, String parent, String path) throws IOException {
-    Path p = new Path(parent,removeLeadingSlash(path));
-    FileStatus[] status = fs.globStatus(p);
-    if (status == null || status.length == 0) {
+  public static FileSelection create(final DrillFileSystem fs, final String parent, final String path) throws IOException {
+    final Path combined = new Path(parent, removeLeadingSlash(path));
+    final FileStatus[] statuses = fs.globStatus(combined);
+    if (statuses == null) {
```

Comment:
We handle null case explicitly here as we cannot create a List down when statuses is null. In case statuses is empty, the delegate create returns `null` which is expected. Can you clarify what you mean by failure?